lntest/itest/lnd_channel-funding_fund_max_test.go

```diff
@@ -0,0 +1,250 @@
+//go:build rpctest
+// +build rpctest
+
+package itest
+
+import (
+	"context"
+	"testing"
+
+	"github.com/btcsuite/btcd/btcutil"
+	"github.com/lightningnetwork/lnd"
+	"github.com/lightningnetwork/lnd/input"
+	"github.com/lightningnetwork/lnd/lnrpc"
+	"github.com/lightningnetwork/lnd/lntest"
+	"github.com/lightningnetwork/lnd/lnwallet/chainfee"
+	"github.com/stretchr/testify/require"
+)
+
+type chanFundMaxTestCase struct {
+	// name is the name of the target test case.
+	name string
+
+	// amount is the amount in the Carol's wallet.
+	amount btcutil.Amount
+
+	// pushAmt is the amount to be pushed to Dave.
+	pushAmt btcutil.Amount
+
+	// feeRate is an optional fee in satoshi/bytes used when opening a
+	// channel.
+	feeRate btcutil.Amount
+
+	// carolBalance is Carol's expected balance in her channel before
+	// subtracting the reserved amount to pay for a commitment transaction.
+	carolBalance btcutil.Amount
+
+	// chanOpenShouldFail denotes if we expect the channel opening to fail.
+	chanOpenShouldFail bool
+}
+
+// testChannelFundMax checks various channel funding scenarios where the user
+// instructed the wallet to use all remaining funds.
+func testChannelFundMax(net *lntest.NetworkHarness,
+	ht *harnessTest) {
+
+	// Create two new nodes that open a channel between each other for these
+	// tests.
+	carol := net.NewNode(ht.t, "Carol", nil)
```

Comment:
nit: using Carol and Dave here messes up all the downstream comments of the opening workflow which always uses Alice and Bob (https://github.com/lightningnetwork/lnd/blob/4b2edc4777f5c6fdc1c0dfd12fe4d8c7048a0825/lntest/harness_net.go#L1073)